### PR TITLE
Update to latest Binaryen

### PR DESCRIPTION
**Pull request by dcodeIO**
This PR accounts for recent changes in Binaryen's C-API and introduces

* v128.**andnot**(a: `v128`, b: `v128`): `v128`
* v128.**swizzle**(a: `v128`, s: `v128`): `v128`
* v128.**load_splat**<`T`>(ptr: `usize`, immOffset?: `usize`, immAlign?: `usize`): `v128`
* v128.**load_ext**<`TFrom`>(ptr: `usize`, immOffset?: `usize`, immAlign?: `usize`): `v128`

along their respective inline-assembler-ish counterparts reflecting Wasm text format, as well as custom section support originally added for https://github.com/AssemblyScript/assemblyscript/pull/890.

Like other relatively fresh instructions, these can't be properly tested yet due to missing implementations in Binaryen's interpreter. Added some commented out tests that we can use as a starting point eventually, though.

### CODE REVIEW

File: std/assembly/builtins.ts [hunk 1]
```diff
@@ -1001,13 +1001,25 @@ export namespace v128 {
...
+  export declare function load(ptr: usize, immOffset?: usize, immAlign?: usize): v128;
 
   // @ts-ignore: decorator
   @unsafe @builtin
-  export declare function store(offset: usize, value: v128, immOffset?: usize, immAlign?: usize): void;
+  export declare function load_splat<T>(ptr: usize, immOffset?: usize, immAlign?: usize): v128;
+
+  // @ts-ignore: decorator
+  @unsafe @builtin
+  export declare function load_ext<TFrom>(ptr: usize, immOffset?: usize, immAlign?: usize): v128;
```
Comment (MaxGraey) [inline std/assembly/builtins.ts]: Why `TFrom` instead `T`?

**Comment by MaxGraey** [comment]
LGTM

File: std/assembly/builtins.ts (hunk 1 above)
Comment (dcodeIO) [reply std/assembly/builtins.ts]: The instruction extends lanes of one type to the respective next larger type (e.g. 8 lanes `u8` to 8 lanes `u16`), so I named it `TFrom` to give a hint that the smaller type is the type argument and the larger type is implicit.

Comment (MaxGraey) [reply std/assembly/builtins.ts]: Got it

**Pull request merged**